### ISSUE TITLE
[CS2] Nodes: Compilation of named classes in expressions is broken

### ISSUE THREAD

**Issue by connec** [jashkenas/coffeescript#4436]
```coffee
new class
# new class {}

new class A
# new (A = )class A {};
```

**Comment by iAmMaxmaxmaximus** [comment]
error to
```coffeescript
new class Cat
```

**Comment by connec** [comment]
That's definitely less esoteric.

**Comment by GeoffreyBooth** [comment]
Fixed by #4569.
